apps/src/lab2/lab2Redux.ts

```diff
@@ -139,6 +146,20 @@ export const setUpWithLevel = createAsyncThunk<
 
     Lab2Registry.getInstance().setAppName(levelProperties.appName);
 
+    // If we are cached, and there is a user app options path because we are in a script
+    // level, then make an async call to the server to find out whether the user is an
+    // instructor, and if they are, then update the user role.  This is needed for the
+    // teacher panel to appear in cached levels.
+    if (getPublicCaching()) {
+      if (payload.userAppOptionsPath) {
+        loadUserAppOptions(payload.userAppOptionsPath).then(result => {
```

Comment:
As discussed offline, this will result in re-fetching this data for every level, but I think that's a fine short term solution as anything more complicated would likely result in creating a new API route. We technically only need one field currently (`isInstructor`) which is only script dependent, but the route itself requires lesson and level ID, so it's probably best to not make any assumptions, and provide all the right data.

Longer term (as we've been discussing) we'll probably want to pull this into a composite user data API that aggregates relevant data across different sources, that we only need to refresh when the user changes. Additionally/alternatively, we may also want to look into having the TeacherPanel component absorb this call, so it can manage fetching the data it needs to present itself.